src/slick.grid.ts

```diff
@@ -3377,24 +3392,45 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
    * @param {Boolean} [suppressSetOverflow] - do we want to suppress the call to `setOverflow`
    */
   setOptions(args: Partial<O>, suppressRender?: boolean, suppressColumnSet?: boolean, suppressSetOverflow?: boolean): void {
+    prepareForOptionsChange();
```

Comment:
Unless this function is local, which I don't think it is, I would assume that you need the `this` here because SlickGrid is now a class and in TypeScript you must always include the `this` (or `super`) in contrary to C# where it's optional
```ts
this.prepareForOptionsChange();
```